doggos/fuzzy_sets/interval_type2_fuzzy_set.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+from typing import Callable, Tuple, NoReturn
+
+
+from doggos.fuzzy_sets.fuzzy_set import FuzzySet
+
+
+class IntervalType2FuzzySet(FuzzySet):
+    """
+    Class used to represent a fuzzy set type II :
+    https://en.wikipedia.org/wiki/Fuzzy_set
+
+    Attributes
+    --------------------------------------------
+    __upper_membership_function : Callable[[float], float]
+        upper membership function, determines the upper degree of belonging to a fuzzy set
+    __lower_membership_function: Callable[[float], float]
+        lower membership function, determines the lower degree of belonging to a fuzzy set
+
+    Methods
+    --------------------------------------------
+    __call__(x: float) -> Tuple[float, float]
+        calculate the degree of belonging to a fuzzy set of an element
```

Comment:
Consider changing "calculate the degree of belonging to a fuzzy set of an element" to "calculate degree of membership of element to interval type II fuzzy set"